docs/docs/create-transformer-plugin.md

```diff
@@ -2,36 +2,272 @@
+module.exports = {
+  plugins: [
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        path: `./src/data/`,
+      },
+    },
+  ],
+}
+```
+
+Now if we query `allFile`...
+
+```graphql
+query {
+  allFile {
+    edges {
+      node {
+        internal {
+          type
+          mediaType
+          description
+          owner
+        }
+      }
+    }
+  }
+}
+```
+
+... We see we have the sourced file.
+
+```json
+{
+  "data": {
+    "allFile": {
+      "edges": [
+        {
+          "node": {
+            "internal": {
+              "contentDigest": "c1644b03f380bc5508456ce91faf0c08",
+              "type": "File",
+              "mediaType": "text/yaml",
+              "description": "File \"src/data/example.yml\"",
+              "owner": "gatsby-source-filesystem"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Transform nodes of type `text/yaml`
+
+Now we can transform the sourced data by hooking into the `onCreateNode` API in `gatsby-node.js`.
+
+If you're following along in an example project, install the following packages:
+
+```shell
+npm install --save js-yaml lodash
+```
+
+Now, in `gatsby-node.js`:
 
 ```javascript:title=gatsby-node.js
-exports.setFieldsOnGraphQLNodeType = require(`./extend-node-type`)
+const jsYaml = require(`js-yaml`)
+
+async function onCreateNode({ node, loadNodeContent }) {
+  // only log for nodes of mediaType `text/yaml`
+  if (node.internal.mediaType !== `text/yaml`) {
+    return
+  }
+
+  const fileContent = await loadNodeContent(node)
+  const parsedContent = jsYaml.load(content)
+}
+
+exports.onCreateNode = onCreateNode
+```
+
+File content:
+
+```text
+- id: Jane Doe
+  bio: Developer based in Somewhere, USA
+- id: John Smith
+  bio: Developer based in Maintown, USA
+```
+
+Parsed YAML content:
+
+```javascript
+;[
+  {
+    id: "Jane Doe",
+    bio: "Developer based in Somewhere, USA",
+  },
+  {
+    id: "John Smith",
+    bio: "Developer based in Maintown, USA",
+  },
+]
+```
+
+Now we'll write a helper function to transform the parsed YAML content into new Gatsby nodes:
+
+```javascript
+function transformObject(obj, id, type) {
+  const yamlNode = {
+    ...obj,
+    id,
+    children: [],
+    parent: node.id,
+    internal: {
+      contentDigest: createContentDigest(obj),
+      type,
+    },
+  }
+  createNode(yamlNode)
+  createParentChildLink({ parent: node, child: yamlNode })
+}
+```
+
+Above, we create a `yamlNode` object with the form expected by the [`createNode` action](/docs/actions/#createNode).
+
+We then create a link between the parent node (file) and the child node (yaml content).
+
+In our updated `gatsby-node.js`, we'll then iterate through the parsed YAML content, using the helper function to transform each into a new node:
+
+```javascript:title=gatsby-node.js
+const jsYaml = require(`js-yaml`)
+const _ = require(`lodash`)
+
+async function onCreateNode({
+  node,
+  actions, // highlight-line
```

Comment:
This highlight-line stuff is _so_ much nicer. Thanks for using it 👍

Reply:
I agree, and I haven't even used it yet! It just seems way easier than squinting my eyes and hoping I identified the correct numbers of lines to highlight haha